ui/src/app/workflows/components/workflows-list/workflows-list.tsx

```diff
@@ -135,6 +136,18 @@ export function WorkflowsList({match, location, history}: RouteComponentProps<an
         if (pagination.limit) {
             params.append('limit', pagination.limit.toString());
         }
+
+        // Add any workflow parameters to the query
+        const workflowProperties = getWorkflowParametersFromQuery(history);
+
+        Object.keys(workflowProperties).forEach(key => {
+            params.append(`parameters[${key}]`, workflowProperties[key]);
+        });
+
+        // Add the sidePanel query parameter if it exists
+        params.append('sidePanel', getSidePanel());
+        params.append('template', queryParams.get('template'));
```

Comment:
I'm not sure why this component is taking an allow-list approach to query parameters, but you could get rid of this extra code by changing `const params = new URLSearchParams()` to `const params = new URLSearchParams(history.location.search);`

I'm not sure what side-effects that'll have, though.

Reply:
We could preserve all the parameters. We might have to try different scenarios to check for side-effects.